insights/specs/default.py

```diff
@@ -311,6 +311,26 @@ def httpd_cmd(broker):
         # https://access.redhat.com/solutions/21680
         return list(ps_httpds)
 
+    @datasource(Mount)
+    def httpd_on_nfs(broker):
+        mnt = broker[Mount]
+        mps = mnt.search(mount_type='nfs4')
+        # get nfs 4.0 mount points
+        nfs_mounts = [m.mount_point for m in mps if m['mount_options'].get("vers") == "4.0"]
+        if nfs_mounts:
+            # get all httpd ps
+            httpd_pids = broker[HostContext].shell_out("pgrep httpd")
+            if httpd_pids:
+                open_nfs_files = 0
+                lsof_cmds = ["lsof -p {}".format(pid) for pid in httpd_pids if pid]
+                # maybe there are thousands open files
+                httpd_open_files = broker[HostContext].shell_out(lsof_cmds)
+                for line in httpd_open_files:
+                    items = line.split()
+                    if len(items) > 8 and items[8].startswith(tuple(nfs_mounts)):
+                        open_nfs_files += 1
+                return {"http_ids": httpd_pids, "nfs_mounts": nfs_mounts, "open_nfs_files": open_nfs_files}
```

Comment:
Datasources defined with `@datasource` that will be invoked with `python -m insights.tools.cat` should return a [DatasourceProvider](https://github.com/RedHatInsights/insights-core/blob/master/insights/core/spec_factory.py#L137) instance so they'll output correctly.